datacube/storage/_rio.py

```diff
@@ -165,8 +164,7 @@ def open(self) -> Iterator[GeoRasterReader]:
 
         try:
             _LOG.debug("opening %s", self.filename)
-            with rasterio.DatasetReader(rasterio.path.parse_path(str(self.filename)),
-                                        sharing=False) as src:
```

Comment:
`sharing=False` is important for Dask use-case, can't just drop it. Pretty sure you can pass it on to `.open`. Keep in mind that `.open` does extra checks with respect to GDAL environment and AWS settings and in the past that caused significant slow-downs. `rasterio` promised to keep supporting direct access to `DatasetReader`, plan is to allow string inputs.

Reply:
I've added `sharing=False` to the `.open` call and everything seems to still work.

DatasetReader appears not accept string inputs as of 1.2.10.